src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -1,12 +1,79 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
+import copy
 
-from . import ApiController, RESTController
+from . import ApiController, RESTController, Task
 from .. import mgr
-from ..security import Scope
+from ..exceptions import DashboardException
+from ..security import Permission, Scope
+from ..services.orchestrator import OrchClient
+from ..services.exception import handle_orchestrator_error
+
+
+def host_task(name, metadata, wait_for=10.0):
+    return Task("host/{}".format(name), metadata, wait_for)
+
+
+def merge_with_orchestrator(mgr_hosts):
```

Comment:
(agreeing with the other points)

> By doing that, I cannot easily use `merge_with_orchestrator` for just getting the orchestrator hosts (well, I can, but I need to pass an empty list, which will be copied into a separate one...).

If you want to just have the orchestrator hosts, then you can can just call the orchestrator.

Reply:
> Not a big fan of functional programming, but here I wonder why not having a method of which does not modify the input list and just provides the orchestrator list of hosts?

The function does not modify input list because of `copy` code. And as @sebastian-philipp pointed out, if we need only hosts in orchestrator, just use the orchestrator client.

Reply:
Sorry, I should switch to `deepcopy` for this.